lib/resources/r4/financial_transaction.yaml

```diff
@@ -291,3 +291,20 @@ fields:
       - display: Currencies
         system: urn:iso:std:iso:4217
         info_link: https://hl7.org/fhir/r4/valueset-currencies.html
+
+- name: Apply First In First Out Extension
+  required: 'No'
+  type: Boolean
+  description: Whether to apply FIFO logic for recurring encounters or not.
+  action: create
+  example: |
+    {
+      "url": "https://fhir-ehr.cerner.com/r4/StructureDefinition/apply-first-in-first-out",
+      "valueBoolean": true
+    }
+  note: >
+    <ul>
+      <li>The default value is boolean value 'true'.</li>
```

Comment:
I think we need notes around how the payment will be applied in CPA when set to true or false:
When the value is 'true' the payment will be posted across existing self-pay balances.
When the value is 'false' the system creates a new recurring encounter and posts the payment to the new self pay balance.

Reply:
Addressed in c339094cbe932d48ad48a1eec08bab902ffe9245

Reply:
Thanks Joseph. Did you see the additional notes changes CPA engineering requested in the comment on https://jira2.cerner.com/browse/RCFHIRAPI-791? :

- When the value is 'true' the payment will be posted across existing financial activity with any outstanding balances using FIFO logic. If there is no existing financial activity, the system will create a new financial activity for current month and post the payment to it. If there are existing financial activity with no outstanding balance, the system will post the payment to the latest month financial activity.
- When the value is 'false' the system will post the payment to the current month financial activity if exists and if not the system creates a new financial activity for current month and post the payment to it.
- I would update "Only supported when the target is Encounter"  to say "Only supported for the Payment Transaction Type with target as Encounter"
- I would also add a note to say "This extension is only supported for recurring encounter and not other types of encounters like inpatient, outpatient, etc."

Reply:
Addressed in be8f1c0e06221800273c9522eaebb20ceb7738c1

Reply:
Added a few corrections in https://github.com/cerner/fhir.cerner.com/commit/be8f1c0e06221800273c9522eaebb20ceb7738c1

Reply:
Addressed in a586dda044536a91d678f12655308908c2ab4f88